src/Zynga/Framework/PgData/V1/PgModel/Cache.hh

```diff
@@ -94,6 +104,33 @@ class Cache implements CacheInterface {
     }
   }
 
+  public function doesWriterOverrideKeyExist<TModelClass as PgRowInterface>(
+    classname<TModelClass> $model,
+    PgWhereClauseInterface $where,
+  ): bool {
+    try {
+      $pgModel = $this->pgModel();
+      if (!$pgModel->allowWriterOverride()) {
+        return false;
+      }
+
+      $cachedRs = $pgModel->reader()->createCachedResultSet($model, $where);
+
+      $cache = $this->getResultSetCache()->getConfig()->getCache();
+
+      $key = $cachedRs->createChecksum().self::WRITER_OVERRIDE_SUFFIX;
+      $writerOverride = $cache->get(new PgWriterOverride(), $key);
+      return $writerOverride !== null;
```

Comment:
should this also check for the value of override in $writerOverride ?

return $writerOverride !== null && $writerOverride->override->get() === true;